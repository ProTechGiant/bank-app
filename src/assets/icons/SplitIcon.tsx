import { Path, Svg } from "react-native-svg";

import { IconProps } from ".";

export function SplitIcon({ height = 24, width = 24, color = "#FF7512" }: IconProps) {
  return (
    <Svg width={width} height={height} viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
      <Path
        d="M15.0516 10.6001L13.4016 8.9751L16.2516 6.1501H14.9516C14.6516 6.1501 14.3932 6.03743 14.1766 5.8121C13.9599 5.58743 13.8516 5.31676 13.8516 5.0001C13.8516 4.68343 13.9642 4.41243 14.1896 4.1871C14.4142 3.96243 14.6849 3.8501 15.0016 3.8501H19.0016C19.3182 3.8501 19.5889 3.96243 19.8136 4.1871C20.0389 4.41243 20.1516 4.68343 20.1516 5.0001V9.0001C20.1516 9.31676 20.0389 9.58743 19.8136 9.8121C19.5889 10.0374 19.3182 10.1501 19.0016 10.1501C18.6849 10.1501 18.4142 10.0418 18.1896 9.8251C17.9642 9.60843 17.8516 9.3501 17.8516 9.0501V7.7501L15.0516 10.6001ZM12.0016 20.1501C11.6849 20.1501 11.4142 20.0374 11.1896 19.8121C10.9642 19.5874 10.8516 19.3168 10.8516 19.0001V12.4501L6.15156 7.7251V9.0251C6.15156 9.34176 6.0389 9.60843 5.81356 9.8251C5.5889 10.0418 5.31823 10.1501 5.00156 10.1501C4.6849 10.1501 4.41423 10.0374 4.18956 9.8121C3.96423 9.58743 3.85156 9.31676 3.85156 9.0001V5.0001C3.85156 4.68343 3.96423 4.41243 4.18956 4.1871C4.41423 3.96243 4.6849 3.8501 5.00156 3.8501H9.00156C9.31823 3.8501 9.5889 3.96243 9.81356 4.1871C10.0389 4.41243 10.1516 4.68343 10.1516 5.0001C10.1516 5.31676 10.0432 5.58743 9.82656 5.8121C9.6099 6.03743 9.34323 6.1501 9.02656 6.1501H7.72656L13.1516 11.5501V19.0001C13.1516 19.3168 13.0389 19.5874 12.8136 19.8121C12.5889 20.0374 12.3182 20.1501 12.0016 20.1501Z"
        fill={color}
      />
    </Svg>
  );
}
