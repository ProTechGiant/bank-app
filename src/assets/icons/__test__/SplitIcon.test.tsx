import { create } from "react-test-renderer";

import { SplitIcon } from "../SplitIcon";

describe("icon/SplitIcon", () => {
  it("renders a SplitIcon", () => {
    const result = create(<SplitIcon />);

    expect(result).toMatchSnapshot();
  });
});
