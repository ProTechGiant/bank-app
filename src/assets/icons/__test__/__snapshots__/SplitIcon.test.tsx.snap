// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon/SplitIcon renders a SplitIcon 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight="24"
  bbWidth="24"
  fill="none"
  focusable={false}
  height={24}
  meetOrSlice={0}
  minX={0}
  minY={0}
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "flex": 0,
        "height": 24,
        "width": 24,
      },
    ]
  }
  vbHeight={24}
  vbWidth={24}
  width={24}
  xmlns="http://www.w3.org/2000/svg"
>
  <RNSVGGroup
    fill={null}
    propList={
      [
        "fill",
      ]
    }
  >
    <RNSVGPath
      d="M6.42188 9.96094L7.22656 8.92188L7.35156 9C9.04688 10.0859 11.1484 12.6406 11.1484 14.4609V18.7578C11.1484 19.3828 11.4531 19.7188 12 19.7188C12.5469 19.7188 12.8438 19.3828 12.8438 18.7578V14.4609C12.8438 12.6406 14.9453 10.0859 16.6406 9L16.7266 8.94531L17.4531 9.94531C17.8281 10.4609 18.3281 10.3125 18.5234 9.70312L19.5781 6.52344C19.7422 6.03125 19.4609 5.64062 18.9375 5.64844L15.5938 5.67188C14.9531 5.67188 14.6641 6.10156 15.0312 6.60938L15.7188 7.55469C14.1172 8.63281 12.3594 10.75 12.0156 12.0078H11.9844C11.6328 10.7422 9.88281 8.64062 8.28125 7.5625L8.95312 6.69531C9.33594 6.20312 9.0625 5.76562 8.42188 5.74219L5.07031 5.61719C4.55469 5.59375 4.26562 5.96875 4.41406 6.46875L5.36719 9.67969C5.54688 10.2969 6.03125 10.4609 6.42188 9.96094Z"
      fill={
        {
          "payload": 4294931730,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
        ]
      }
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;
